include/shad/core/vector.h

```diff
@@ -0,0 +1,1044 @@
+//===------------------------------------------------------------*- C++ -*-===//
+//
+//                                     SHAD
+//
+//      The Scalable High-performance Algorithms and Data Structure Library
+//
+//===----------------------------------------------------------------------===//
+//
+// Copyright 2018 Battelle Memorial Institute
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy
+// of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef INCLUDE_SHAD_CORE_VECTOR_H_
+#define INCLUDE_SHAD_CORE_VECTOR_H_
+
+#include <cstdlib>
+
+#include "shad/data_structures/abstract_data_structure.h"
+#include "shad/distributed_iterator_traits.h"
+#include "shad/runtime/runtime.h"
+
+namespace shad {
+
+namespace impl {
+/// @brief Distributed vector without resizing capabilities.
+/// TODO change the descriptions
```

Comment:
Please add the missing ref (or remove the comment Section XX)

Reply:
Done